File: app/src/main/java/com/quattro/models/ServicioAuxiliarModel.java
package com.quattro.models;

import android.database.Cursor;
import BaseDatos.ServicioAuxiliar;

public class ServicioAuxiliarModel {


    //region CONSTRUCTORES

    public ServicioAuxiliarModel() { }

    public ServicioAuxiliarModel(Cursor cursor){
        Linea = cursor.getString(cursor.getColumnIndexOrThrow("Linea"));
        Servicio = cursor.getString(cursor.getColumnIndexOrThrow("Servicio"));
        Turno = cursor.getInt(cursor.getColumnIndexOrThrow("Turno"));
        LineaAuxiliar = cursor.getString(cursor.getColumnIndexOrThrow("LineaAuxiliar"));
        ServicioAuxiliar = cursor.getString(cursor.getColumnIndexOrThrow("ServicioAuxiliar"));
        TurnoAuxiliar = cursor.getInt(cursor.getColumnIndexOrThrow("TurnoAuxiliar"));
        Inicio = cursor.getString(cursor.getColumnIndexOrThrow("Inicio"));
        LugarInicio = cursor.getString(cursor.getColumnIndexOrThrow("LugarInicio"));
        Final = cursor.getString(cursor.getColumnIndexOrThrow("Final"));
        LugarFinal = cursor.getString(cursor.getColumnIndexOrThrow("LugarFinal"));
    }

    public ServicioAuxiliarModel (ServicioAuxiliarModel auxiliar){
        FromModel(auxiliar);
        Nuevo = true;
    }

    //endregion


    //region CAMPOS PÚBLICOS

    public Boolean Nuevo = false;

    public Boolean Modificado = false;

    //endregion


    //region MÉTODOS PÚBLICOS

    public ServicioAuxiliar ToServicioAuxiliar(){
        ServicioAuxiliar auxiliar = new ServicioAuxiliar();
        auxiliar.setLinea(Linea);
        auxiliar.setServicio(Servicio);
        auxiliar.setTurno(Turno);
        auxiliar.setLineaAuxiliar(LineaAuxiliar);
        auxiliar.setServicioAuxiliar(ServicioAuxiliar);
        auxiliar.setTurnoAuxiliar(TurnoAuxiliar);
        auxiliar.setInicio(Inicio);
        auxiliar.setFinal(Final);
        auxiliar.setLugarInicio(LugarInicio);
        auxiliar.setLugarFinal(LugarFinal);
        return auxiliar;
    }


    public void FromModel(ServicioAuxiliarModel servicio){
        Linea = servicio.getLinea();
        Servicio = servicio.getServicio();
        Turno = servicio.getTurno();
        LineaAuxiliar = servicio.getLineaAuxiliar();
        ServicioAuxiliar = servicio.getServicioAuxiliar();
        TurnoAuxiliar = servicio.getTurnoAuxiliar();
        Inicio = servicio.getInicio();
        LugarInicio = servicio.getLugarInicio();
        Final = servicio.getFinal();
        LugarFinal = servicio.getLugarFinal();
    }


    public Boolean EsIgual(ServicioAuxiliarModel auxiliar){
        return Linea.equals(auxiliar.getLinea()) &&
                Servicio.equals((auxiliar.getServicio())) &&
                Turno == auxiliar.getTurno() &&
                LineaAuxiliar.equals(auxiliar.getLineaAuxiliar()) &&
                ServicioAuxiliar.equals(auxiliar.getServicioAuxiliar()) &&
                TurnoAuxiliar == auxiliar.getTurnoAuxiliar();
    }

    //endregion


    //region PROPIEDADES


    // LINEA
    private String Linea = "";
    public String getLinea() {
        return Linea;
    }
    public void setLinea(String linea) {
        if(!linea.equals(Linea)) {
            Linea = linea;
            Modificado = true;
        }
    }


    // SERVICIO
    private String Servicio = "";
    public String getServicio() {
        return Servicio;
    }
    public void setServicio(String servicio) {
        if (!servicio.equals(Servicio)) {
            Servicio = servicio;
            Modificado = true;
        }
    }


    // TURNO
    private int Turno = 0;
    public int getTurno() {
        return Turno;
    }
    public void setTurno(int turno) {
        if (turno != Turno) {
            Turno = turno;
            Modificado = true;
        }
    }


    // LÍNEA AUXILIAR
    private String LineaAuxiliar;
    public String getLineaAuxiliar() {
        return LineaAuxiliar;
    }
    public void setLineaAuxiliar(String lineaAuxiliar) {
        if (!lineaAuxiliar.equals(LineaAuxiliar)) {
            LineaAuxiliar = lineaAuxiliar;
            Modificado = true;
        }
    }


    // SERVICIO AUXILIAR
    private String ServicioAuxiliar;
    public String getServicioAuxiliar() {
        return ServicioAuxiliar;
    }
    public void setServicioAuxiliar(String servicioAuxiliar) {
        if (!servicioAuxiliar.equals(ServicioAuxiliar)){
            ServicioAuxiliar = servicioAuxiliar;
            Modificado = true;
        }
    }


    // TURNO AUXILIAR
    private int TurnoAuxiliar;
    public int getTurnoAuxiliar() {
        return TurnoAuxiliar;
    }
    public void setTurnoAuxiliar(int turnoAuxiliar) {
        if (turnoAuxiliar != TurnoAuxiliar){
            TurnoAuxiliar = turnoAuxiliar;
            Modificado = true;
        }
    }


    // INICIO
    private String Inicio = "";
    public String getInicio() {
        return Inicio;
    }
    public void setInicio(String inicio) {
        if (!inicio.equals(Inicio)) {
            Inicio = inicio;
            Modificado = true;
        }
    }


    // FINAL
    private String Final = "";
    public String getFinal() {
        return Final;
    }
    public void setFinal(String aFinal) {
        if (!aFinal.equals(Final)) {
            Final = aFinal;
            Modificado = true;
        }
    }


    // LUGAR INICIO
    private String LugarInicio = "";
    public String getLugarInicio() {
        return LugarInicio;
    }
    public void setLugarInicio(String lugarInicio) {
        if (!lugarInicio.equals(LugarInicio)) {
            LugarInicio = lugarInicio;
            Modificado = true;
        }
    }


    // LUGAR FINAL
    private String LugarFinal = "";
    public String getLugarFinal() {
        return LugarFinal;
    }
    public void setLugarFinal(String lugarFinal) {
        if (!lugarFinal.equals(LugarFinal)) {
            LugarFinal = lugarFinal;
            Modificado = true;
        }
    }


    //endregion

}

File: app/src/main/java/com/quattroid/data/models/ServicioAuxiliarModel.kt
/*
 * AnderSoft - Open Source Software
 * Licencia GPL 3.0 - 2025
 * Visite https://www.gnu.org/licenses/gpl-3.0.html para más detalles.
 */

package com.quattroid.data.models

import android.content.ContentValues
import android.database.Cursor


class ServicioAuxiliarModel : BaseModel {


    var servicioId: Int = 0
        set(value) {
            field = value
            modified = true
        }

    var Servicioid: Int = 0
        set(value) {
            field = value
            modified = true
        }


    var linea: String = ""
        set(value) {
            field = value
            modified = true
        }

    var servicio: String = ""
        set(value) {
            field = value
            modified = true
        }

    var turno: Int = 0
        set(value) {
            field = value
            modified = true
        }

    var lineaAuxiliar: String = ""
        set(value) {
            field = value
            modified = true
        }

    var servicioAuxiliar: String = ""
        set(value) {
            field = value
            modified = true
        }

    var turnoAuxiliar: Int = 0
        set(value) {
            field = value
            modified = true
        }

    var inicio: String = ""
        set(value) {
            field = value
            modified = true
        }

    var lugarInicio: String = ""
        set(value) {
            field = value
            modified = true
        }

    var final: String = ""
        set(value) {
            field = value
            modified = true
        }

    var lugarFinal: String = ""
        set(value) {
            field = value
            modified = true
        }

    constructor() {}

    constructor(cursor: Cursor) {
        id = cursor.getInt(cursor.getColumnIndexOrThrow("_id"))
        servicioId = cursor.getInt(cursor.getColumnIndexOrThrow("ServicioId"))
        linea = cursor.getString(cursor.getColumnIndexOrThrow("Linea"))
        servicio = cursor.getString(cursor.getColumnIndexOrThrow("Servicio"))
        turno = cursor.getInt(cursor.getColumnIndexOrThrow("Turno"))
        lineaAuxiliar = cursor.getString(cursor.getColumnIndexOrThrow("LineaAuxiliar"))
        servicioAuxiliar = cursor.getString(cursor.getColumnIndexOrThrow("ServicioAuxiliar"))
        turnoAuxiliar = cursor.getInt(cursor.getColumnIndexOrThrow("TurnoAuxiliar"))
        inicio = cursor.getString(cursor.getColumnIndexOrThrow("Inicio"))
        lugarInicio = cursor.getString(cursor.getColumnIndexOrThrow("LugarInicio"))
        final = cursor.getString(cursor.getColumnIndexOrThrow("Final"))
        lugarFinal = cursor.getString(cursor.getColumnIndexOrThrow("LugarFinal"))
        modified = false
    }

    fun clone(): ServicioAuxiliarModel {
        val nuevoServicioAuxiliar = ServicioAuxiliarModel()
        nuevoServicioAuxiliar.fromModel(this)
        return nuevoServicioAuxiliar
    }

    fun fromModel(other: ServicioAuxiliarModel) {
        val wasModified = modified
        id = other.id
        servicioId = other.servicioId
        linea = other.linea
        servicio = other.servicio
        turno = other.turno
        lineaAuxiliar = other.lineaAuxiliar
        servicioAuxiliar = other.servicioAuxiliar
        turnoAuxiliar = other.turnoAuxiliar
        inicio = other.inicio
        lugarInicio = other.lugarInicio
        final = other.final
        lugarFinal = other.lugarFinal
        modified = wasModified
    }


    fun getContentValues(): ContentValues {
        val valores = ContentValues()
        valores.put("ServicioId", servicioId)
        valores.put("Linea", linea)
        valores.put("Servicio", servicio)
        valores.put("Turno", turno)
        valores.put("LineaAuxiliar", lineaAuxiliar)
        valores.put("ServicioAuxiliar", servicioAuxiliar)
        valores.put("TurnoAuxiliar", turnoAuxiliar)
        valores.put("Inicio", inicio)
        valores.put("LugarInicio", lugarInicio)
        valores.put("Final", final)
        valores.put("LugarFinal", lugarFinal)
        return valores
    }

}